src/contract/client.ts

```diff
@@ -20,15 +32,73 @@ import { processSpecEntryStream } from './utils';
  * @param {ClientOptions} options see {@link ClientOptions}
  */
 export class Client {
+  static async deploy<T = Client>(
+    args: Record<string, any> | null,
+    options: MethodOptions &
+      Omit<ClientOptions, "contractId"> & {
+        wasmHash: Buffer | string;
+        salt: Buffer;
+        format?: "hex" | "base64";
+      }
+  ): Promise<AssembledTransaction<T>> {
+    let spec = await specFromWasmHash(
+      options.wasmHash,
+      options,
+      options.format
+    );
+    const wasmHashBuffer =
+      typeof options.wasmHash === "string"
+        ? Buffer.from(options.wasmHash, options.format)
+        : (options.wasmHash as Buffer);
+    let constructorArgs: xdr.ScVal[] = args
+      ? spec.funcArgsToScVals(CONSTRUCTOR_FUNC, args)
+      : [];
+    let address = new Address(options.publicKey!);
+    let contractIdPreimage =
+      xdr.ContractIdPreimage.contractIdPreimageFromAddress(
+        new xdr.ContractIdPreimageFromAddress({
+          salt: options.salt,
+          address: address.toScAddress(),
+        })
+      );
+    let contractId = Address.contract(
+      hash(contractIdPreimage.toXDR())
+    ).toString();
+    console.log("contractId", contractId);
+    let func = xdr.HostFunction.hostFunctionTypeCreateContractV2(
+      new xdr.CreateContractArgsV2({
+        constructorArgs,
+        contractIdPreimage,
+        executable:
+          xdr.ContractExecutable.contractExecutableWasm(wasmHashBuffer),
+      })
+    );
+    let operation = Operation.invokeHostFunction({
```

Comment:
[`Operation.createCustomContract`](https://stellar.github.io/js-stellar-sdk/Operation.html#.createCustomContract) should clean some of this code up :+1:

Reply:
But I need the constructorArgs?

Reply:
The [latest version](https://github.com/stellar/js-stellar-base/releases/tag/v13.0.0-beta.1) has those! I think I need to look into the docgen job.